### ISSUE TITLE
Bug in Zstd.compressUsingDict method

### ISSUE THREAD

**Issue by ctasada** [luben/zstd-jni#126]
Hi,

I'm upgrading a project from `1.4.0-1` to `1.4.5-4` and some of my test start breaking. Analyzing the issue I found that's related with the `compressUsingDict` method. This method was `native` on my old dependency, but now there's some Java code wrapping the native method.

```
   public static long compressUsingDict (byte[] dst, int dstOffset, byte[] src, int srcOffset, int length, byte[] dict, int level) {
        ZstdCompressCtx ctx = new ZstdCompressCtx();
        try {
            ctx.setLevel(level);
            ctx.loadDict(dict);
            return (long) ctx.compress(dst, src);
        } finally {
            ctx.close();
        }
    }
```
As we can see in the code, `dstOffset`, `srcOffset` and `length` are ignored. Once I moved my code to copy arrays around, it worked as expected.

Ideally this method should respect the expectations set by the documentation and old behaviour.

**Comment by luben** [comment]
Hmm, good catch, looks like copy/paste error.

**Comment by luben** [comment]
I pushed a commit and will release it a little bit later today

**Comment by ctasada** [comment]
Thanks a lot for the fast reply! really appreciated :)

**Comment by luben** [comment]
v1.4.5-5 is on maven with the fix

**Comment by ctasada** [comment]
Hi @luben, 

Sorry for the late delay, but I went on vacations a few days.

I have updated my code and I still have the same issue.

Checking the commit https://github.com/luben/zstd-jni/commit/2662fbdc320ce482a24c20b8fcac8b1d5b79fe33 I think there may be an issue. Sadly I cannot manage to compile in my environment, so I cannot validate.

```
return (long) ctx.compressByteArray(dst, dstOffset - dstOffset, dst.length, src, srcOffset, length);
```
probably should be
```
return (long) ctx.compressByteArray(dst, dstOffset, dst.length - dstOffset, src, srcOffset, length);
```
like in the `compressFastDict` method.

The way the code is right now, is the same as saying that `dstOffset` is always `0`

Note: Changing my code to use `compressFastDict` instead of `compressUsingDict` solves my issues, so that seems to confirm the bug.
